docs/basics/networkbehaviour.md

```diff
@@ -108,3 +108,130 @@ If you override the virtual 'OnDestroy' method it is important to alway invoke t
 
 `NetworkBehaviour` handles other destroy clean up tasks and requires that you invoke the base `OnDestroy` method to operate properly.
 :::
+
+### NetworkBehaviour Pre-Spawn Synchronization
+
+There could be scenarios where you need to include additional configuration data or use a `NetworkBehaviour` to configure some non-netcode related component (or the like) prior to a `NetworkObject` being spawned. This can be particularly critical if you want specific settings applied prior to `NetworkBehaviour.OnNetworkSpawn` being invoked. When a client is synchronizing with an existing network session, this can become problematic as messaging requires a client to be fully synchronized before you really know "it is safe" to send the message and even if you send a message there is the latency involved in the whole process that might not be convenient and could require additional specialized code to account for this.
+
+`NetworkBehaviour.OnSynchronize` allows you to write and read custom serialized data during the `NetworkObject` serialization process.  
+
+There are two cases where `NetworkObject` synchronization occurs:
+- When dynamically spawning a `NetworkObject`.
+- When a client is being synchronized after connection approval 
+  - i.e. Full synchronization of the `NetworkObject`s and scenes.
+
+:::info
+If you haven't already become familiar with the [`INetworkSerializable` interface](../advanced-topics/serialization/inetworkserializable.md), then you might read up on that before proceeding as `NetworkBehaviour.OnSynchronize` as it follows a very similar usage pattern.
+:::
+
+#### Order of Operations When Dynamically Spawning:<br />
+The following provides you with an outline of the order of operations that occur during `NetworkObject` serialization when dynamically spawned.
+
+Server-Side:
+- `GameObject` with `NetworkObject` component is instantiated.
+- The `NetworkObject` is spawned. 
+  - For each associated `NetworkBehaviour` component, `NetworkBehaviour.OnNetworkSpawn` is invoked.
+- The `CreateObjectMessage` is generated 
+  - `NetworkObject` state is serialized.
+  - `NetworkVariable` state is serialized.
+  - `NetworkBehaviour.OnSynchronize` is invoked for each `NetworkBehaviour` component.
+    - If this method is not overridden then nothing is written to the serialization buffer.
+- The `CreateObjectMessage` is sent to all clients that are observers of the `NetworkObject`.
+<br />
+
+Client-Side:
+- The `CreateObjectMessage` is received
+  - `GameObject` with `NetworkObject` component is instantiated.
+  - `NetworkVariable` state is deserialized and applied.
+  - `NetworkBehaviour.OnSynchronize` is invoked for each `NetworkBehaviour` component.
+    - If this method is not overridden then nothing is read from the serialization buffer.
+- The `NetworkObject` is spawned
+  - For each associated `NetworkBehaviour` component, `NetworkBehaviour.OnNetworkSpawn` is invoked.
+
```

Comment:
I'm a little on the fence about this section. Normally, the style guide enforces active voice, but there are some scenarios when translating to active voice makes the content less clear. Let me know what you think 🙂

- - -

The following outlines the order of operations that occur during `NetworkObject` serialization when dynamically spawned.

Server-side:
- `GameObject` with `NetworkObject` component instantiates.
- The server spawns the `NetworkObject`. 
  - For each associated `NetworkBehaviour` component, the server invokes `NetworkBehaviour.OnNetworkSpawn`.
- The server generates `CreateObjectMessage`. 
  - The server serializes the `NetworkObject` state.
  - The server serializes the `NetworkVariable` state.
  - The server invokes `NetworkBehaviour.OnSynchronize` for each `NetworkBehaviour` component.
    - If you don't override this method, the server doesn't write anything to the serialization buffer.
- The server sends `CreateObjectMessage` to all clients that are observers of the `NetworkObject`.
<br />

Client-side:
- The client receives `CreateObjectMessage`.
  - `GameObject` with `NetworkObject` component instantiates.
  - The client deserializes and applies the `NetworkVariable` state.
  - The client invokes `NetworkBehaviour.OnSynchronize` for each `NetworkBehaviour` component.
    - If you don't override this method, the client doesn't read anything from the serialization buffer.
- The client spawns the `NetworkObject`.
  - For each associated `NetworkBehaviour` component, the client invokes `NetworkBehaviour.OnNetworkSpawn`.